### PR TITLE
RC-208 Dedicated address to override feedback email notifications

### PR DESCRIPTION
**Pull request by GrimStal**
# Ticket

[RC-208](https://dbmedialab.atlassian.net/browse/RC-208)

# What need to be noticed in solution

Made an ability to send email notifications about feedbacks to dedicated email addresses instead of article authors emails

### CODE REVIEW

File: src/app/services/website/common/validateAndUpdate.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export default function(name: string, data : any) : Promise <Website> {
 		return Promise.reject(error);
 	}
 
-	return Promise.all([])
+	return Promise.resolve()
```
Comment (liordoe) [inline src/app/services/website/common/validateAndUpdate.ts]: ?

File: src/app/services/website/common/validateAndUpdate.ts [hunk 2]
```diff
@@ -48,10 +48,23 @@ function validateSchema(data : any) {
...
+	dataArrays.forEach((item: string) => {
+		if (item in data && (!data[item] || !Array.isArray(data[item]))) {
+			throw new SchemaValidationError(`Invalid website data: ${item} must be an array`);
+		}
+	});
+
+	if (data.feedbackEmailOverride) {
+		data.feedbackEmailOverride.forEach((item) => {
+			const test = new RegExp(
+				'^[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*' +
```
Comment (liordoe) [inline src/app/services/website/common/validateAndUpdate.ts]: set this to constants 'cause it's some kind of 'magic numbers'

File: src/base/ValidationCustomValidations.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ export interface ICustomValidations {
 
 const customValidations = {
 	isEmail: function(schema: String, v:any) {
-		if (!/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/i.test(v)) {
+		if (!/^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/i.test(v)) {
```
Comment (liordoe) [inline src/base/ValidationCustomValidations.ts]: and here

File: src/admin/components/website/modalParts/WebsiteFeedbackEmailOverride.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+			}
+
+			return false;
+		}
+	}
+
+	onDelete (index) {
+		const {feedbackEmailOverride: override} = this.props;
+		if (index >= 0) {
+			const feedbackEmailOverride = override.asMutable();
```
Comment (sergeyvlakh) [inline src/admin/components/website/modalParts/WebsiteFeedbackEmailOverride.tsx]: \>_>

File: src/admin/components/website/modalParts/WebsiteFeedbackEmailOverride.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+			feedbackEmailOverride.splice(index, 1);
```
Comment (sergeyvlakh) [inline src/admin/components/website/modalParts/WebsiteFeedbackEmailOverride.tsx]: use pop(index) if you only delete 1 element by index

Comment (GrimStal) [reply src/admin/components/website/modalParts/WebsiteFeedbackEmailOverride.tsx]: pop remove only last element, not by index

**Pull request merged**